chalk-integration/src/lowering.rs

```diff
@@ -64,16 +67,39 @@ const SELF: &str = "Self";
 const FIXME_SELF: &str = "__FIXME_SELF__";
 
 impl<'k> Env<'k> {
-    fn lookup(&self, name: Identifier) -> LowerResult<NameLookup> {
+    fn lookup_type(&self, name: Identifier) -> LowerResult<TypeLookup> {
```

Comment:
Similarly here. Name of `lookup_type` can only return struct, parameter, or not struct.

Reply:
Right -- this looks up something that is intended to be a type.